### PR TITLE
Increase test coverage for the mapping parser

### PR DESCRIPTION
**Pull request by jbarnoud**
Target 100% coverage for `map_input.py`.

### CODE REVIEW

**Comment by jbarnoud** [comment]
Should be good to go.

File: vermouth/map_input.py [hunk 1]
```diff
@@ -92,16 +94,32 @@ def read_mapping(lines):
...
+        ]))
+        for from_atom, to_atoms in mapping.items()
     }
-    for bead_weights in weights.values():
-        for from_atom, count in bead_weights.items():
-            bead_weights[from_atom] = 1 / count
+    for atom_weights in pre_weights.values():
+        total = sum(atom_weights.values())
+        for to_atom, count in atom_weights.items():
+            atom_weights[to_atom] = count / total
```
Comment (pckroon) [inline vermouth/map_input.py]: `atom_weights[to_atom] /= total`?

File: vermouth/tests/test_map_input.py [hunk 2]
```diff
@@ -181,15 +191,152 @@
...
+        [ atoms ]
+        0 X1{0} A{0} B{0}
+        1 X2{0} C{0} D{0}
+    """)
+
+    mappings = collections.defaultdict(lambda: collections.defaultdict(dict))
+
+    force_fields_from = ['ff{}'.format(i) for i in range(4)]
+    force_fields_to = force_fields_from + ['only_to']
+    force_fields_from += ['only_from']
```
Comment (pckroon) [inline vermouth/tests/test_map_input.py]: `force_fields_from = force_fields_from + ['only_from']` to make it symmetric with the line above?

File: vermouth/tests/test_map_input.py [hunk 3]
```diff
@@ -221,4 +368,98 @@ def test_generate_self_mapping():
...
+                node['atomname'] = atomname
+        force_fields.append(force_field)
+
+        expected.append((ff_name, (ff_name, )))
+
+    mappings = vermouth.map_input.generate_all_self_mappings(force_fields)
+    found = [(from_ff, tuple(to_ff.keys())) for from_ff, to_ff in mappings.items()]
+
+    # In python <= 3.5, dicts are not ordered.
+    assert sorted(found) == sorted(expected)
```
Comment (pckroon) [inline vermouth/tests/test_map_input.py]: I'd turn it into a set comparison, but this also works I think

**Pull request merged**